tuist/tuist
# Preserve xcuserdata when re-generating projects

Resolves #1003 

### Short description 📝
- The project generation process was replacing the entire `.xcodeproj` and `.xcworkspace` directories
- This lead to wiping all previous contents within those directories
- This included a nested `xcuserdata` directory which is usually gitignored and holds the Xcode UI state


### Solution 📦

- To resolve this, only a the individual files are replaced rather than the entire directory

### Implementation 👩‍💻👨‍💻

- [x] Update project generator
- [x] Update workspace generator

### Test Plan 🛠:

- Run `tuist generate` within any fixture
- Open the generated project
- Navigate around, set breakpoints
- Re-order one of the source files (i.e. to cause a re-generation)
- Close Xcode
- Run `tuist generate` again
- Open the generated project
- Verify the last viewed file is opened automatically, breakpoints, the recently viewed files list are all still there
